model/character.go

```diff
@@ -106,43 +104,41 @@ func (player *CharacterStats) DeterminePhysicalDamage() {
 		}
 	}
 
-	player.PhysicalDamage = int(damage)
+	player.PhysicalDamage = damage
 }
 
-// ChangeXP changes XP player from XPAmount, could be negative, return true if leveled up
-func (player *CharacterStats) ChangeXP(XPAmount int) bool {
-	player.CurrentXP = player.CurrentXP + XPAmount
+// ChangeXP changes XP player from XPAmount, could be negative, return true if leveled up.
+func (player *CharacterStats) ChangeXP(xpAmount int) bool {
+	player.CurrentXP += xpAmount
 	// Since we change XP, check if level changes
 	return player.DetermineLevel()
 }
 
-// ChangeGold changes amount of gold of player from GoldAmount, could be negative
-func (player *CharacterStats) ChangeGold(GoldAmount int) {
+// ChangeGold changes amount of gold of player from GoldAmount, could be negative.
+func (player *CharacterStats) ChangeGold(goldAmount int) {
 	// TODO: add some random elements
-	player.CurrentGold = int(player.CurrentGold) + GoldAmount
+	player.CurrentGold += goldAmount
 	player.ComputeWeight()
 }
 
 // DetermineLevel check player currentXP and increase level if necessary
 // You can't loose levels even if you lost XP (by design). Returns true if
-// player leveled up
+// player leveled up.
 func (player *CharacterStats) DetermineLevel() bool {
 	for i, requiredXP := range xpTable {
-		//fmt.Printf("Current XP %d", player.CurrentXP)
 		if player.CurrentXP >= requiredXP {
 			// we are still above threshold, continue
-			//fmt.Printf("%d continue", i)
+
 			continue
 		} else {
-			//fmt.Printf("%d stop", i)
 			// we are bellow next threshold, that's our level
 			if i > player.Level {
 				// only change level if it's greater than current
 				// there could be effects removing XP but I don't want to affect level
 
 				// increase PointsToSpend by 2 per new levels
 				// it shouldn't be more than 2 points each time when the game is finished/balanced
-				player.PointsToSpend = player.PointsToSpend + 2*(i-player.Level)
+				player.PointsToSpend += 2 * (i - player.Level)
```

Comment:
```suggestion
				player.PointsToSpend += 2 * (i - player.Level)
```

> model/character.go:143:5: assignOp: replace `player.PointsToSpend = player.PointsToSpend + 2*(i-player.Level)` with `player.PointsToSpend += 2*(i-player.Level)` (gocritic)
                                player.PointsToSpend = player.PointsToSpend + 2*(i-player.Level)